content/realtime/channels/channel-parameters/deltas.textile

```diff
@@ -0,0 +1,207 @@
+
+If a delta is generated and it results in a difference that is not appreciably smaller than the original message, or is even larger than the original message (which can happen if successive messages are completely different), then the delta will not be sent and clients will receive the original, unprocessed message. Therefore, the sequence of messages that will be delivered to a client for any given channel will be a combination of regular messages and delta-compressed messages.
+
+A channel subscriber can experience a discontinuity in the sequence of messages it receives on a given channel for the following reasons:
+
+* The connection can drop, and there will be a discontinuity of the client is unable to reconnect within the two-minute window it is allowed to preserve connection continuity.
+* The outbound connection might have been rate-limited, which causes some messages to be dropped
+* There might have been some internal error in the Ably system which leads to the server being unable to preserve continuity on the channel. 
+
+In these cases, the service indicates the discontinuity to the client, together with the reason, and this is usually visible to the subscriber in a channel @UPDATE@ event. 
+
+If a subscriber has a delta-mode subscription and the channel in question experiences a discontunity, then a non-delta message will be delivered to the client as the first message after the discontinuity. This ensures that lost messages do not prevent the client from reconstituting messages from deltas.
+
+h2(#using-deltas). Using deltas
+
+h3(#using-deltas-ably). Via an Ably library
+
+The most common way to subscribe to Ably channels is via a realtime connection, using an Ably realtime library.
+
+For many libraries this requires no change on the part of the caller except to specify the @delta@ "channel parameter":/realtime/channels/channel-parameters/overview when subscribing to the channel. In some libraries, the @vcdiff@ delta decoding library is excluded from the default library distribution in order to avoid bloating the library. In these cases, it is also necessary to supply the delta decoder plugin when instancing the Ably library.
+
+
+```[jsall]
+  const vcdiffPlugin = require('{{vcdiff-plugin}}')
+  const realtime = new Ably.Realtime({key: '{{API_KEY}}', plugins: {vcdiffDecoder: vcdiffDecoder}});
```

Comment:
@tomczoink is this the one you're building a JSBin for? @QuintinWillison tagging you here as well if you have the answer with regards to the variable naming.

Reply:
In the [origin readme](https://github.com/ably-forks/vcdiff-decoder#installation-from-npm-for-nodejs) I've named it just `vcdiff`, however `vcdiffPlugin` is more descriptive so you could just keep that.

Matt's right about `vcdiffDecoder` - this code seems to be a mashup from a few sources, perhaps. If it's being called `vcdiffPlugin` then that's what you need to pass in `plugins`, though the dictionary key is also wrong - it should be `vcdiff`... see https://github.com/ably/ably-js/pull/656/files

Also, I would prefer to use `const` in preference to `var` as that makes the intention clearer and is probably available to most JavaScript programmers now, however others may have stronger opinions there.

Reply:
See https://github.com/ably/docs/pull/880#issuecomment-637452883

Reply:
@QuintinWillison @MarkWoulfeAbly after this work is done, I think you should both have a think about how we move documentation code examples from "I hope this is right" to "this is tested and we know works because it passes CI".  I have seen a number of companies solve this problem in unique ways, which I am happy to talk you through.  However, as the team grows, and the SDKs evolve, I think it's important we tackle this so that our examples are robst.

Reply:
See @mattheworiordan's [internal thread](https://ably-real-time.slack.com/archives/C0127MBHDT9/p1591136055101000) on this, where I've responded.